Full Yellow Local - Fix Rootcheck tests in 4.2

## Description

This issue is part of the https://github.com/wazuh/wazuh-qa/issues/1516. After the research #1493, it has been detected errors in `rootcheck` tests for Wazuh manager in Linux systems. In order to sanitize current tests is necessary to fix rootcheck tests.


##  Tests results

### Pipeline parameters

|  Jenkins branch  | QA branch   |  Version 	|   Revision |
|- |- |- |- |
|  4.2 |   1516-4.2.0-full-green | 4.2  |  1.1493 | 

### Execution results

This test is only available for Linux managers.


|   |  **Local** | **Jenkins**  |  
|:--:|:--: |:--:|
|**Linux Manager**|  🔴   |   :red_circle:    | 

#### Local
- [**R1**](https://github.com/wazuh/wazuh-qa/files/6733516/r1-rootcheck-manager-centos8.log)
- [**R2**](https://github.com/wazuh/wazuh-qa/files/6733518/r2-rootcheck-manager-centos8.log)
- [**R3**](https://github.com/wazuh/wazuh-qa/files/6733519/r3-rootcheck-manager-centos8.log)

#### Jenkins
- [**R1**](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/881/)
- [**R2**](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/925/)
- [**R3**](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/954/)

### Description of the error

```
_______ test_rootcheck[check_updates:False, check_delete:True, 1_agents] _______

get_configuration = {'apply_to_modules': ['test_rootcheck'], 'metadata': {'agents_number': 1, 'check_delete': True, 'check_updates': False}, 'sections': [{'elements': [{'disabled': {'value': 'no'}}], 'section': 'rootcheck'}], 'tags': ['all']}
configure_environment = None, restart_service = None, clean_alert_logs = None

...
        for agent in agents:
            rows = retrieve_rootcheck_rows(agent.id)
            db_string = [row[3] for row in rows]
    
            logs_string = [':'.join(x.split(':')[2:]) for x in
>                          agent.rootcheck.messages_list]
E           AttributeError: 'NoneType' object has no attribute 'messages_list'

test_rootcheck/test_rootcheck.py:149: AttributeError
---------------------------- Captured stdout setup -----------------------------
2021-06-29T07:34:00.1624952040
Restarting wazuh-manager (via systemctl):  [  OK  ]
----------------------------- Captured stdout call -----------------------------
Stopping wazuh-manager (via systemctl):  [  OK  ]
----------------------------- Captured stderr call -----------------------------
Exception in thread Thread-040rootcheck:
...
FileNotFoundError: [Errno 2] No such file or directory: '/home/vagrant/wazuh-qa/deps/wazuh_testing/wazuh_testing/tools/../data/040'
_______ test_rootcheck[check_updates:False, check_delete:True, 3_agents] _______
```

Rootcheck test works correctly in master branch. This error could be related to the changes in `agent_simulator`  module in 4.2 branch.


### Development

This fix should be developed in a branch from 4.2, and it must be merged in the epic issue full-green branch `1516-4.2.0-full-green`

In order to finish this issue the following tasks should be fulfilled:

- [x] Full green in rootcheck tests for every supported OS:
  - [x] test_rootcheck  :red_circle: 
- [x] Documentation of any important change done for these tests or the used tools
- [x] Merge with `1516-4.2.0-full-green`

--- a/deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py
@@ -648,7 +648,7 @@ class Agent:
     def init_rootcheck(self):
         """Initialize rootcheck module."""
         if self.rootcheck is None:
-            self.rootcheck = Rootcheck(self.rootcheck_sample, self.name, self.id)
+            self.rootcheck = Rootcheck(os = self.os, agent_name = self.name, agent_id = self.id, rootcheck_sample = self.rootcheck_sample)
 
     def init_fim(self):
         """Initialize fim module."""
@@ -934,8 +934,6 @@ class SCA:
 
 
 class Rootcheck:
-    def __init__(self, os, agent_name, agent_id, rootcheck_sample=None):
-        self.os = os
     """This class allows the generation of rootcheck events.
 
     Creates rootcheck events by sequentially repeating the events of a sample file file.
@@ -945,7 +943,8 @@ class Rootcheck:
         agent_id (str): Id of the agent.
         rootcheck_sample (str): File with the rootcheck events that are going to be used.
     """
-    def __init__(self, agent_name, agent_id, rootcheck_sample=None):
+    def __init__(self, os, agent_name, agent_id, rootcheck_sample=None):
+        self.os = os
         self.agent_name = agent_name
         self.agent_id = agent_id
         self.rootcheck_tag = 'rootcheck'
@@ -966,7 +965,7 @@ class Rootcheck:
             line = fp.readline()
             while line:
                 if not line.startswith("#"):
-                    msg = "{0}:{1}:{2}".format(self.ROOTCHECK_MQ, self.ROOTCHECK, line.strip("\n"))
+                    msg = "{0}:{1}:{2}".format(self.rootcheck_mq, self.rootcheck_tag, line.strip("\n"))
                     self.messages_list.append(msg)
                 line = fp.readline()
 

